static/app/components/core/button/index.tsx

```diff
@@ -354,29 +354,42 @@ export const StyledButton = styled(
     ) => {
       // XXX: There may be two forwarded refs here, one potentially passed from a
       // wrapped Tooltip, another from callers of Button.
-      const ref = mergeRefs([forwardRef, forwardRefAlt]);
+      const ref = mergeRefs(forwardRef, forwardRefAlt);
 
       // Get component to use based on existence of `to` or `href` properties
       // Can be react-router `Link`, `a`, or `button`
       if (to) {
         return (
-          <Link {...props} ref={ref} to={to} replace={replace} disabled={disabled} />
+          <Link
+            {...props}
+            ref={ref as React.Ref<HTMLAnchorElement>}
```

Comment:
Why do we need to as it now?

Reply:
yeah its two types and it doesn't like that

Reply:
oh this is becuase of button and link button probably right

Reply:
what we’d technically want is a `HTMLAnchorElement` if `to` is specified, and a `HTMLButtonElement` otherwise. We would need a discriminated union type for that. I think we can make this work after we’ve removed `forwardRef`

Reply:
Well we just need to finish splitting out Button and LinkButton